benchopt/plotting/html/static/benchmark.js

```diff
@@ -0,0 +1,175 @@
+  var filter = new Array();
+  for (item = 0; item < ll_item.length; item++) {
+    filter.push(
+      document
+        .getElementById("select" + ll_item[item].replace(/\s/g, ""))
+        .value.toUpperCase()
+    ); // Add dropdown value on top of the array (order is necessary)
+  }
+  var table = document.getElementById("summary");
+  var tr = table.getElementsByTagName("tr");
+  for (i = 0; i < tr.length; i++) {
+    tr[i].style.display = "";
+    td = tr[i].getElementsByTagName("td")[2];
+    if (td) {
+      // do not display elements from table not in filter
+      for (fil = 0; fil < filter.length; fil++) {
+        if (
+          td.innerHTML.toUpperCase().indexOf(filter[fil]) > -1 &&
+          tr[i].style.display !== "none"
+        ) {
+          tr[i].style.display = "";
+        } else {
+          tr[i].style.display = "none";
+        }
+      }
+    }
+  }
+}
+
+/**
+ * Display subinfo in the table
+ * Toggle between + and - icon button
+ */
+function displaymore(id, loop_index) {
+  var x = document.getElementById("subinfo" + loop_index);
+  $(id).find("svg").toggleClass("fa-plus-circle fa-minus-circle");
+  if (x.style.display === "none") {
+    x.style.display = "block";
+  } else {
+    x.style.display = "none";
+  }
+}
+
+// handle sidebar navigation
+var openbtn = document.getElementById("open");
+if (openbtn) {
+  var closebtn = document.getElementById("close");
+  openbtn.style.display = "none"; // only display close button is sidenav opened
+
+  // function to open the sidebar navigation
+  function openSideNav() {
+    navbar = document.getElementById("sidenav");
+    navbar.style.visibility = "visible";
+    main = document.getElementById("main");
+    if ($(window).width() < 900) {
+      // mobile adaptativity
+      navbar.style.height = "100%";
+      main.style.marginLeft = "0";
+    } else {
+      navbar.style.height = "1000px";
+      main.style.marginLeft = "200px";
+    }
+    navbar.style.opacity = "1";
+    openbtn.style.display = "none"; // hide open button
+    closebtn.style.display = ""; // show close button
+  }
+
+  // default display depending on device
+  if ($(window).width() > 900) {
+    openSideNav();
+  } else {
+    closeSideNav();
+  }
+
+  // function to close the sidebar navigation
+  function closeSideNav() {
+    navbar = document.getElementById("sidenav");
+    main = document.getElementById("main");
+    navbar.style.visibility = "hidden";
+    navbar.style.opacity = "0";
+    navbar.style.height = "0";
+    if ($(window).width() > 900) {
+      // adapt to devices
+      main.style.marginLeft = "0px";
+    }
+    openbtn.style.display = ""; // show open button
+    closebtn.style.display = "none"; // hide close button
+  }
+}
+
+// dialog box to get path of files we would like to remove
+$(function () {
+  $("#dialogRm").dialog({
+    autoOpen: false, // only open when needed
+  });
+
+  // click on delete button in the dialog box
+  $("#trashBtn").click(function () {
```

Comment:
To be consistent, I would rather use the `onclick` attribute of the HTML button and define this one as a callback?

Reply:
It should be the preferred way to handle click events actually (recently learned that)
See [this as ref and more explanations](https://stackoverflow.com/questions/12627443/jquery-click-vs-onclick)

Reply:
Do you think i would be possible to define the function inside `click` outside and add the callback as it is done for the open and close button?